Update SOC Manifest for Caliptra 2.0

mlvisaya opened the pull request:
- Remove Image Source
- Add Image Identifier
- Add ComponentInfo fields from PLDM
- Add Load Address and Entry point

swenson commented on auth-manifest/README.md:
```diff
@@ -3,37 +3,50 @@
+| Image Hash              | 48           | SHA2-384 hash of a SOC image                                                                                                                                                                                                                                                                                                                                         ...
+| Image Identifier        | 4            | Unique value selected by the vendor to distinguish between component images.<br />IMEs should be sorted by an ascending order of Image Identifier within an IMC.                                                                                                                                                                                                     ...
+| Flags                   | 4            | A set of 32-bit flags specific to the image.<br />**Bit0**: If set, the image hash will not be verified otherwise the metadata image hash will be compared against the calculated hash of the image.<br />**Bit1**: If set, it indicates that the image is a MCU Runtime image, otherwise it indicates a SOC Image.                                                  ...
+| Image Load Address High | 4            | The higher 4-bytes of the 64-bit AXI address where the image will be loaded for verification and execution or staging.                                                                                                                                                                                                                                               ...
+| Image Load Address Low  | 4            | The lower 4-bytes of the 64-bit AXI address where the image will be loaded for verification and execution or staging                                                                                                                                                                                                                                                 ...
```
nit: "4-byte" (with hyphen) is for when it is an adjective, but "4 bytes" is for when it is a noun.

```suggestion
| Image Load Address High | 4            | The higher 4 bytes of the 64-bit AXI address where the image will be loaded for verification and execution or staging.                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
| Image Load Address Low  | 4            | The lower 4 bytes of the 64-bit AXI address where the image will be loaded for verification and execution or staging                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
```

swenson reviewed (commented):
Minor nits only

mlvisaya committed:
Update SOC Manifest for Caliptra 2.0
- Updated the "Preamble" section to include detailed descriptions of the Version, SVN, Flags, and other fields.
- Reformatted field tables for improved readability and consistency.
- Added new fields such as IMC Vendor ECC Signature, IMC Vendor LMS Signature, IMC Owner ECC Signature, and IMC Owner LMS Signature.
- Expanded "Image Metadata Entry" section with new fields like Image Identifier, Flags, Image Load Address, Classification, Version Number, and Version String.
- Revised "Image Metadata Collection" section for clarity and added a new table format.

mhatrevi commented on auth-manifest/README.md:
```diff
@@ -3,37 +3,50 @@
+| Field                   | Size (bytes) | Description                                                                                                                                                                                                                                                                                                                                                          ...
+| ----------------------- | ------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ...
+| Image Hash              | 48           | SHA2-384 hash of a SOC image                                                                                                                                                                                                                                                                                                                                         ...
+| Image Identifier        | 4            | Unique value selected by the vendor to distinguish between component images.<br />IMEs should be sorted by an ascending order of Image Identifier within an IMC.                                                                                                                                                                                                     ...
```
This is not a requirement. Runtime will sort the entries internally for faster lookup. The only requirement is that these Ids should be unique.

mlvisaya replied:
got it. ill update

mhatrevi commented on auth-manifest/README.md:
```diff
@@ -3,37 +3,50 @@
 
 - ### **Image Metadata Collection**
+
 The Image Metadata Collection (IMC) is a collection of Image Metadata entries (IME). Each IME has a hash that matches a SOC images. The manifest vendor and owner private keys sign the IMC. The Preamble holds the IMC signatures. The manifest IMC vendor signatures are optional and are validated only if the FLAGS field Bit 0 = 1. Up to sixteen image hashes are supported.
```
This should be 127.

mhatrevi reviewed (changes requested):
Please see the feedback.

mlvisaya committed:
Addressed feedback

- remove requirement that images are sorted by image identifier
- update number of images supported

mlvisaya committed:
Merge branch 'main-2.x' into mlvisaya/doc-update
